local/local_provider.go

```diff
@@ -63,10 +70,86 @@ func (r *Local) CreateDatabase(db *crd.Database) (string, error) {
 	return db.Name, nil
 }
 
+const (
+	defaultLocalRDSPVSize = "1Gi"
+	awsPVCWaitPerionSec   = time.Second * 10
+)
+
+func (r *Local) createPVC(name, namespace string) error {
+	newPVC := false
+
+	pvcName := name + "-pvc"
+	pvc, err := r.kc.CoreV1().PersistentVolumeClaims(namespace).Get(pvcName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		// we got an error and it's not the NotFound, let's crash
+		return err
+	}
+	if errors.IsNotFound(err) {
+		// Deployment seems to be empty, let's assume it means we need to create it
+		pvc = &corev1.PersistentVolumeClaim{}
+		newPVC = true
+	}
+
+	pvc.Name = pvcName
+	pvc.ObjectMeta = metav1.ObjectMeta{
+		Name: pvcName,
+		Labels: map[string]string{
+			"app": name,
+		},
+	}
+
+	pvc.Annotations = map[string]string{
+		"team":       "dev-prod",
+		"repository": "https://github.com/sorenmat/k8s-rds",
+	}
+
+	pvc.Spec = corev1.PersistentVolumeClaimSpec{
+		AccessModes: []corev1.PersistentVolumeAccessMode{
+			"ReadWriteOnce",
+		},
+
+		Resources: corev1.ResourceRequirements{
+
+			Requests: corev1.ResourceList{
+				"storage": resource.MustParse(defaultLocalRDSPVSize),
+			},
+		},
+	}
+
+	if newPVC {
+		log.Printf("creating pvc %v", pvcName)
+		_, err = r.kc.CoreV1().PersistentVolumeClaims(namespace).Create(pvc)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Printf("updating pvc %v", pvcName)
+		_, err = r.kc.CoreV1().PersistentVolumeClaims(namespace).Update(pvc)
+		if err != nil {
+			return err
+		}
+	}
+
+	// waiting arbitrary amount of time for PVCgit  to bound the volume
+	time.Sleep(awsPVCWaitPerionSec)
+
+	return nil
+}
+
 func (r *Local) DeleteDatabase(db *crd.Database) error {
 	// delete the database instance
-	err := r.kc.AppsV1().Deployments(db.Namespace).Delete(db.Name, &metav1.DeleteOptions{})
-	return err
+	if err := r.kc.AppsV1().Deployments(db.Namespace).Delete(db.Name,
+		&metav1.DeleteOptions{}); err != nil {
+		return err
```

Comment:
I guess we shouldn't return... if we deleted the deployment and the controller crashed, then we would never be able to delete the PVC. Maybe just log the error and try to delete the PVC?